### ISSUE TITLE
Full Site Editing: only load FSE for sites that use Gutenberg

### ISSUE THREAD

**Issue by vindl** [Automattic/wp-calypso#35178]
FSE can't work with Classic editor so its functionality shouldn't be enabled if the user hasn't opted in.

One problem that I foresee with this is that our current opt-in settings are per user per site - subset of users that have opted in will be able to edit the header and footer templates, but other users of the same site won't. It's get even weirder for the front end since those users won't have a way to customize these areas at all.

In this case it seems like it would make more sense it opt-in was on per site level. Instead of changing our opt-in libraries, I think we should just force loading Gutenberg for all users on sites that have `full-site-editing` sticker applied. We should be able to contain this change to a function that determines whether Gutenberg should be loaded on Dotcom.

cc @mmtr @kwight for thoughts on opt in functionality too.
Related issue: https://github.com/Automattic/wp-calypso/issues/32833 - Once we force loading the opt-out actions no longer make sense and should be removed.

**Comment by noahtallen** [comment]
I feel like this is related: https://github.com/orgs/Automattic/projects/90#card-24984423

**Comment by vindl** [comment]
Removing blocker status since this is not a blocker for Call for Testing.

**Comment by mmtr** [comment]
@noahtallen I think this will be fixed by D31361-code, right?

**Comment by noahtallen** [comment]
> I think we should just force loading Gutenberg for all users on sites that have full-site-editing sticker applied

Yeah, it should resolve this. I'm not sure what behavior would be with an existing user who opted out because I don't have any older accounts :P I imagine they would still get Gutenberg if they get a site with FSE. Not really an issue at the minute since existing users can't see FSE... Just not sure how it will handle that case.
